src/Markdown/Parser.elm

```diff
@@ -729,15 +751,26 @@ whenPreviousWasNotBody =
         , orderedListBlock False
         , heading |> Advanced.backtrackable
         , htmlParser
-
-        -- TODO re-enable this once the table parser handles rows
-        --, TableParser.parser |> Advanced.backtrackable |> map Table
+        , tableRowIfTableStarted revStmts.rawBlocks |> Advanced.backtrackable
+        , TableParser.headerParser |> Advanced.backtrackable |> map (\(Markdown.Table.TableHeader headers) -> Table (Markdown.Table.Table headers []))
         , plainLine
         ]
-        |> map (\block revStmts -> Loop (possiblyMergeBlocks revStmts block))
+        |> map (\block -> Loop (possiblyMergeBlocks revStmts block))
     ]
 
 
+tableRowIfTableStarted : List RawBlock -> Parser RawBlock
+tableRowIfTableStarted revStmts =
+    case revStmts of
+        (Table (Markdown.Table.Table headers body)) :: _ ->
+            TableParser.bodyRowParser (List.length headers)
+                -- We always fill in the whole table so that we don't have an impossible state of a table row without a table header
```

Comment:
I know this is a bit weird when we join them by just discarding the former element, but it felt worse to have a `TableRow` `RawBlock` that wasn't guaranteed to be valid via types.